<a href="https://colab.research.google.com/github/priyanka-sidhuraj/127/blob/main/c133_Priyanka.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [44]:
import csv
import plotly.express as px

rows = [] #all th data

with open("main.csv", "r") as f:
  csvreader = csv.reader(f)
  for row in csvreader: 
    rows.append(row)

headers = rows[0]
planet_data_rows = rows[1:]



In [45]:
headers[0] = "row_num"

solar_system_planet_count = {}

for planet_data in planet_data_rows:
  if solar_system_planet_count.get(planet_data[11]):
    solar_system_planet_count[planet_data[11]] += 1 #checking the count
  else:
    solar_system_planet_count[planet_data[11]] = 1

max_solar_system = max(solar_system_planet_count, key=solar_system_planet_count.get)

# lis = ['1', '100', '111', '2']
#max(lis)
#'111'


print("Solar system which has maximum planets out of all the solar systems we have discovered so far!",max_solar_system)

Solar system which has maximum planets out of all the solar systems we have discovered so far! KOI-351


In [46]:
temp_planet_data_rows = list(planet_data_rows)

for planet_data in temp_planet_data_rows:
  planet_mass = planet_data[3]
  if planet_mass.lower() == "unknown": #unknown
    planet_data_rows.remove(planet_data)
    continue
  else:
    planet_mass_value = planet_mass.split(" ")[0]
    planet_mass_ref = planet_mass.split(" ")[1]
    if planet_mass_ref == "Jupiters":
      planet_mass_value = float(planet_mass_value) * 317.8
    planet_data[3] = planet_mass_value

  planet_radius = planet_data[7]
  if planet_radius.lower() == "unknown":
    planet_data_rows.remove(planet_data)
    continue
  else:
    planet_radius_value = planet_radius.split(" ")[0]
    planet_radius_ref = planet_radius.split(" ")[2]
    if planet_radius_ref == "Jupiter":
      planet_radius_value = float(planet_radius_value) * 11.2
    planet_data[7] = planet_radius_value


koi_351_planets = []
for planet_data in planet_data_rows:
  if max_solar_system == planet_data[11]:
    koi_351_planets.append(planet_data)

print(len(koi_351_planets))

7


The Value of G (Gravitational Constant) is 6.674e-11

Mass of Earth = 5.972e+24

Radius of Earth = 6371000

In [47]:
temp_planet_data_rows = list(planet_data_rows)


planet_masses = []
planet_radiuses = []
planet_names = []
for planet_data in planet_data_rows:
  planet_masses.append(planet_data[3])
  planet_radiuses.append(planet_data[7])
  planet_names.append(planet_data[1])
planet_gravity = []
for index, name in enumerate(planet_names):
  gravity = (float(planet_masses[index])*5.972e+24)  * 6.674e-11/ (float(planet_radiuses[index])*float(planet_radiuses[index])*6371000*6371000)
  planet_gravity.append(gravity)

fig = px.scatter(x=planet_radiuses, y=planet_masses)
fig.show()

Our Earth’s gravity is 9.8 m/s, and we as humans are accustomed to it.

In order for us to exist on any other planet, the gravity should be close to what we have here.

Mars has a gravity of 3.711 m/s and Moon has a gravity of 1.62 m/s.

In [48]:
print(headers)

['row_num', 'name', 'light_years_from_earth', 'planet_mass', 'stellar_magnitude', 'discovery_date', 'planet_type', 'planet_radius', 'orbital_radius', 'orbital_period', 'eccentricity', 'solar_system_name', 'planet_discovery_method', 'planet_orbital_inclination', 'planet_density', 'right_ascension', 'declination', 'host_temperature', 'host_mass', 'host_radius']


In [49]:
planet_type_values = []
for planet_data in planet_data_rows:
  planet_type_values.append(planet_data[6])

print(list(set(planet_type_values)))

['Super Earth', 'Terrestrial', 'Gas Giant', 'Neptune-like']


Now that we have the types of planets that are out there, let’s understand these terms -

Neptune-like => These planets are like neptune! They are big in size and have rings around them. They are also made of Ice.

Super-Earth => These are the planets that have mass greater than earth but smaller than that of Neptune! (Neptune is 17 times Earth)

Terrestrial => It is a planet that is composed primarily of silicate rocks or metals (Like Earth, Mars)

Gas Giant => There are the planets that are composed of Gas (Hydrogen and Heliu

In [50]:
suitable_planets = []
for planet_data in planet_data_rows:
  if planet_data[6].lower() == "terrestrial" or planet_data[6].lower() == "super earth":
    suitable_planets.append(planet_data)

print(len(suitable_planets))

1485


In [51]:
planet_masses = []
planet_radiuses = []
planet_types = []

for planet_data in planet_data_rows:
  planet_masses.append(planet_data[3])
  planet_radiuses.append(planet_data[7])
  planet_types.append(planet_data[6])

fig = px.scatter(x=planet_radiuses, y=planet_masses, color=planet_types)
fig.show()

In [52]:
low_gravity_planets = []
for index, gravity in enumerate(planet_gravity):
  if gravity > 12: #9.8
    low_gravity_planets.append(planet_data_rows[index])

print(len(low_gravity_planets))

1799


In [53]:
suitable_planets = []
for planet_data in low_gravity_planets:
  if planet_data[6].lower() == "terrestrial" or planet_data[6].lower() == "super earth":
    suitable_planets.append(planet_data)

print(len(suitable_planets))

820


In [54]:
print(headers)

['row_num', 'name', 'light_years_from_earth', 'planet_mass', 'stellar_magnitude', 'discovery_date', 'planet_type', 'planet_radius', 'orbital_radius', 'orbital_period', 'eccentricity', 'solar_system_name', 'planet_discovery_method', 'planet_orbital_inclination', 'planet_density', 'right_ascension', 'declination', 'host_temperature', 'host_mass', 'host_radius']


Orbital radius is the distance of the
planet from the host star.
Orbital period is the time the planet
takes to complete one orbit around its
Sun!

In [55]:
temp_suitable_planets = list(suitable_planets)

for planet_data in temp_suitable_planets:
  if planet_data[8].lower() == "unknown":
    suitable_planets.remove(planet_data)

for planet_data in suitable_planets:
  if planet_data[9].split(" ")[1].lower() == "days":
    planet_data[9] = float(planet_data[9].split(" ")[0]) #Days
  else:
    planet_data[9] = float(planet_data[9].split(" ")[0])*365 
    #Years notation should be converted into days by multiplying it with 365

    
  planet_data[8] = float(planet_data[8].split(" ")[0])

orbital_radiuses = []
orbital_periods = []

for planet_data in suitable_planets:
  orbital_radiuses.append(planet_data[8])
  orbital_periods.append(planet_data[9])

fig = px.scatter(x=orbital_radiuses, y=orbital_periods)
fig.show()

Here, we can see that most of the
suitable planets lie within 2
astronomical units, which is great!
1. If a planet is too close to its
sun, we can say that the planet
will be too hot for us to survive.
2. If a planet is too far away from
its sun, we can say that the
planet will be too cold for us to
survive.
We call this the Goldilock Zone.
Goldilock Zone is the habitable zone
where the planet is more likely to
have just the right conditions to
sustain life.
For us, we are at the very beginning
of the Goldilock Zone whereas Mars
is at the end of it.
Earth is 1AU from the Sun and Mars
is 1.5AU from the Sun.
Some studies suggest on average,
any planet that lies within 0.38 - 2 AU
is likely to be habitable.


In [56]:
goldilock_planets = list(suitable_planets) #We will leave suitable planet list as it is

temp_goldilock_planets = list(suitable_planets) 
for planet_data in temp_goldilock_planets:
  if planet_data[8] < 0.38 or planet_data[8] > 2:
    goldilock_planets.remove(planet_data)

print(len(suitable_planets))
print(len(goldilock_planets))

417
11


The list of suitable planets gives us
417 and out of those, only 11 are in
Goldilock Zone!

Our Earth revolves around the sun at
30km/s. Similarly, our solar system
revolves around the center of the
Milky Way galaxy at the speed of
200km/s.
Given this data, we can
assume that if a planet revolves at the
speed of 200km/s, we can survive
that. We are already surviving such
high speeds!
That means that the planet that’s in
the Goldilock zone would take
anywhere from 30 to 50 days to
complete it’s 1 year (1 revolution
around its sun). It would be 6-7 times
faster than us.


In [57]:
planet_speeds = []
for planet_data in suitable_planets:
  distance = 2 * 3.14 * (planet_data[8] * 1.496e+9) #to convert our orbital radius from AU to KM. 2 pi r
  time = planet_data[9] * 86400 #convreting into seconds
  speed = distance / time
  planet_speeds.append(speed)

speed_supporting_planets = list(suitable_planets) #We will leave suitable planet list as it is

temp_speed_supporting_planets = list(suitable_planets)

for index, planet_data in enumerate(temp_speed_supporting_planets):
  if planet_speeds[index] > 200:
    speed_supporting_planets.remove(planet_data)

print(len(speed_supporting_planets))

5
